test/dist_model_parallel_test.py

```diff
@@ -0,0 +1,138 @@
+import torch
```

Comment:
Wasn't the plan to have this in the [examples](https://github.com/pytorch/examples) repository? Why is this a unit test instead?

Reply:
We need RPC module tests as well. I would first like to check with @soumith on whether this is a reasonable example, and whether there are any APIs that we need to revise. And also waiting for the dist optimizer to be landed (it's landing now).